src/DualQuaternion.jl

```diff
@@ -4,6 +4,14 @@ struct DualQuaternion{T<:Real} <: Number
   q0::Quaternion{T}
   qe::Quaternion{T}
   norm::Bool
+  function DualQuaternion{T}(q0::Quaternion, qe::Quaternion, norm::Bool) where T <: Real
```

Comment:
I think we need this to ensure the message is always raised:
```suggestion
  function DualQuaternion{T}(q0::Quaternion{T}, qe::Quaternion{T}, norm::Bool) where T <: Real
```

Reply:
On the current `master`, we have `DualQuaternion{T}(q0, qe, norm)` method that accepts `Any` type, and we should remove `::Quaternion` rather than adding type parameter `{T}`.

```julia
julia> methods(DualQuaternion{Float64})
# 10 methods for type constructor:
[1] (::Type{T})(x::AbstractChar) where T<:Union{AbstractChar, Number} in Base at char.jl:50
[2] (::Type{T})(x::Base.TwicePrecision) where T<:Number in Base at twiceprecision.jl:266
[3] DualQuaternion{T}(q0::Quaternion) where T<:Real in Quaternions at /home/hyrodium/.julia/dev/Quaternions/src/DualQuaternion.jl:17
[4] DualQuaternion{T}(dq::DualQuaternion) where T<:Real in Quaternions at /home/hyrodium/.julia/dev/Quaternions/src/DualQuaternion.jl:9
[5] DualQuaternion{T}(x::Real) where T<:Real in Quaternions at /home/hyrodium/.julia/dev/Quaternions/src/DualQuaternion.jl:26
[6] DualQuaternion{T}(d::DualNumbers.Dual) where T<:Real in Quaternions at /home/hyrodium/.julia/dev/Quaternions/src/DualQuaternion.jl:20
[7] DualQuaternion{T}(d1::DualNumbers.Dual, d2::DualNumbers.Dual, d3::DualNumbers.Dual, d4::DualNumbers.Dual) where T<:Real in Quaternions at /home/hyrodium/.julia/dev/Quaternions/src/DualQuaternion.jl:10
[8] DualQuaternion{T}(d1::DualNumbers.Dual, d2::DualNumbers.Dual, d3::DualNumbers.Dual, d4::DualNumbers.Dual, n::Bool) where T<:Real in Quaternions at /home/hyrodium/.julia/dev/Quaternions/src/DualQuaternion.jl:10
[9] (::Type{T})(x::T) where T<:Number in Core at boot.jl:772
[10] DualQuaternion{T}(q0, qe, norm) where T<:Real in Quaternions at /home/hyrodium/.julia/dev/Quaternions/src/DualQuaternion.jl:4
```

Reply:
>I think we need this to ensure the message is always raised:

If we add `function` inside `struct`, the default constructor will be disabled. So there is no problem with the message?

Reply:
I have removed the `::Quaterion` type limitation in the constructor function, and it now has the same methods as on the current `master`. I have confirmed these methods with

```julia
for s in sort(string.(methods(DualQuaternion))) println(s) end
for s in sort(string.(methods(DualQuaternionF64))) println(s) end
```